collector/collector.go

```diff
@@ -239,6 +224,42 @@ func (c *Collector) Collect(metrics chan<- prometheus.Metric) {
 	metrics <- prometheus.MustNewConstMetric(c.dbUp, prometheus.GaugeValue, up, c.dbName)
 }
 
+func (c *Collector) ensureConnection() error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if c.pingFail {
+		// ping has failed, continually return err
+		return fmt.Errorf("ping is failing: verify DSN is correct and DB2 is running properly")
+	}
+
+	// mainly here so unit tests work
+	if c.db != nil {
+		return nil
+	}
```

Comment:
Could we check this portion at the first of this function? Feels more idiomatic to return early whenever possible. Not entirely sure if it'll cause issues with the test files but something I think we should aim for.

Reply:
could we potentially add a comment in the code to explain why this check is done later perhaps?